File: Data Structures & Algorithms/jump-game-ii/submission-1.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        for(int i=1;i<nums.size();i++){
            nums[i]+=i;
            nums[i]=max(nums[i],nums[i-1]);
        }
        int id=0;int ans=0;
        while(id<nums.size()-1){
            id=nums[id];ans++;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/k-closest-points-to-origin/submission-1.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        int[][] ans=new int[k][2];
        Queue<List<Integer>> q=new PriorityQueue<List<Integer>>((a,b)->a.get(0)-b.get(0));
        for(int i=0;i<points.length;i++){
            q.add(List.of(points[i][0]*points[i][0]+points[i][1]*points[i][1], points[i][0], points[i][1]));
        }
        for(int i=0;i<k;i++){
            List<Integer> el=q.peek();
            q.poll();
            ans[i][0]=el.get(1);
            ans[i][1]=el.get(2);
        }
        return ans;
    }
}
